server/src/controllers/AuthController.js

```diff
@@ -151,28 +159,70 @@ module.exports = {
     }
   },
 
-  async verifyToken(req, res) {
+  async verifyToken(req, res, next) {
     const errors = validationResult(req);
 
     if (!errors.isEmpty()) {
       res.status(422).json({ errors: errors.array() });
       return;
     }
 
-    const { token } = req.body;
+    const token = req.headers.authorization.split(" ")[1];
 
     try {
-      jwt.verify(token, config.privateKey);
-      res.status(200).json({
-        status: true,
+      jwt.verify(token, config.privateKey, async (err, decoded) => {
+        if(!err){
+          next();
+        }else{
+          res.status(401).json({
+            status: false,
+            error: err.message
+          });
+        }
       });
     } catch (error) {
       res.status(403).json({
         status: false,
+        error: err.message
       });
     }
   },
 
+  async refreshAuthTokens(req, res){
+    try {
+      const { refreshToken, uid } = req.body;
+
+      if(!refreshToken || !uid){
+        return res.status(401).json({ error: "Data missing" });
+      }
+
+      const user = await User.findById(uid);
+
+      jwt.verify(refreshToken, config.refreshPrivateKey, async (err, decoded) => {
+        if(!err){
+          const newToken = jwt.sign({ user: user }, config.privateKey, {
+            expiresIn: ACCESS_TOKEN_EXPIRE_TIME,
+          });
+
+          const newRefreshToken = jwt.sign({user: user}, config.refreshPrivateKey, {
+            expiresIn: REFRESH_TOKEN_EXPIRE_TIME
+          });
+
+          res.status(200).json({
+            token: newToken,
+            refreshToken: newRefreshToken
+          })
+        }else{
+          console.log(e.message);
+          return res.status(401).json({ error: err.message });
+        }
+      });
+    } catch (e) {
+      console.log(e.message);
+      return res.status(401).json({ error: e.message });
```

Comment:
If this try catch fails, it will be due to some error from Mongoose.findByID, since we have already caught the JWT errors one scope above. Likewise, the error response should then be 500.